infoschema/tables_test.go

```diff
@@ -1218,19 +1201,34 @@ func (s *testTableSuite) TestStmtSummaryHistoryTable(c *C) {
 	tk.MustExec("insert into test_summary    values(2, 'b')")
 	tk.MustExec("insert into TEST_SUMMARY VALUES(3, 'c')")
 	tk.MustExec("/**/insert into test_summary values(4, 'd')")
-	tk.MustQuery(`select stmt_type, schema_name, table_names, index_names, exec_count, sum_cop_task_num, avg_total_keys,
-		max_total_keys, avg_processed_keys, max_processed_keys, avg_write_keys, max_write_keys, avg_prewrite_regions,
-		max_prewrite_regions, avg_affected_rows, query_sample_text
-		from information_schema.statements_summary_history
-		where digest_text like 'insert into test_summary%'`,
-	).Check(testkit.Rows("Insert test test.test_summary <nil> 4 0 0 0 0 0 2 2 1 1 1 insert into test_summary values(1, 'a')"))
+
+	sql := "select stmt_type, schema_name, table_names, index_names, exec_count, sum_cop_task_num, avg_total_keys," +
+		"max_total_keys, avg_processed_keys, max_processed_keys, avg_write_keys, max_write_keys, avg_prewrite_regions," +
+		"max_prewrite_regions, avg_affected_rows, query_sample_text " +
+		"from information_schema.statements_summary_history " +
+		"where digest_text like 'insert into `test_summary`%'"
+	tk.MustQuery(sql).Check(testkit.Rows("Insert test test.test_summary <nil> 4 0 0 0 0 0 2 2 1 1 1 insert into test_summary values(1, 'a')"))
 
 	tk.MustExec("set global tidb_stmt_summary_history_size = 0")
 	tk.MustQuery(`select stmt_type, schema_name, table_names, index_names, exec_count, sum_cop_task_num, avg_total_keys,
 		max_total_keys, avg_processed_keys, max_processed_keys, avg_write_keys, max_write_keys, avg_prewrite_regions,
 		max_prewrite_regions, avg_affected_rows, query_sample_text, plan
 		from information_schema.statements_summary_history`,
 	).Check(testkit.Rows())
+
+	tk.MustExec("set global tidb_enable_stmt_summary = 0")
+	tk.MustExec("drop table if exists `table`")
+	tk.MustExec("set global tidb_stmt_summary_history_size = 1")
+	tk.MustExec("set global tidb_enable_stmt_summary = 1")
+	tk.MustExec("create table `table`(`insert` int)")
+	tk.MustExec("select `insert` from `table`")
+
+	sql = "select digest_text from information_schema.statements_summary_history;"
+	tk.MustQuery(sql).Check(testkit.Rows(
+		"select `insert` from `table`",
+		"create table `table` ( `insert` int )",
+		"set global `tidb_enable_stmt_summary` = ?",
+	))
```

Comment:
hi @morgo, is this test suitable for that issue?

Reply:
Yes. I had assumed (my mistake) that you were also fixing https://github.com/pingcap/tidb/issues/22388 - I have verified in your branch that it is not fixed yet though:

```sql
mysql> create binding for select * from t where a >=0 and a > 10 using select * from t ignore index(idx) where a >=0 and a > 10;
ERROR 1105 (HY000): line 1 column 23 near "-ada7-48ca-98cf-b424c3e140eb.t IGNORE INDEX (idx) WHERE a >= 0 AND a > 10" 
mysql> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v4.0.0-beta.2-2175-gb0a9f5fef
Edition: Community
Git Commit Hash: b0a9f5fefcce52cfa6055f5ddae4ef5fdec8611f
Git Branch: pr/22893
UTC Build Time: 2021-02-25 02:40:16
GoVersion: go1.13.15
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
1 row in set (0.00 sec)
```

I think it is fine and we can disregard my previous comment. Issue #22388 can be solved separately.

Reply:
ok, I will take a look at this issue. BTW, can you please help me review this PR? 🤣